test/integration/smoke/test_router_dnsservice.py

```diff
@@ -0,0 +1,268 @@
+            cls.network_offering,
+            cls.service_offering,
+            cls.account
+        ]
+
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.api_client, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.cleanup = []
+
+
+    def tearDown(self):
+        try:
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+
+    def test_router_common(self):
+        """Performs common router tests and returns router public_ips"""
+
+        routers = list_routers(
+            self.apiclient,
+            account=self.account.name,
+            domainid=self.account.domainid
+        )
+
+        self.assertEqual(
+            isinstance(routers, list),
+            True,
+            "Check for list routers response return valid data"
+        )
+
+        self.assertTrue(
+            len(routers) >= 1,
+            "Check list router response"
+        )
+
+        router = routers[0]
+
+        self.assertEqual(
+            router.state,
+            'Running',
+            "Check list router response for router state"
+        )
+
+        public_ips = list_publicIP(
+            self.apiclient,
+            account=self.account.name,
+            domainid=self.account.domainid,
+            zoneid=self.zone.id
+        )
+
+        self.assertEqual(
+            isinstance(public_ips, list),
+            True,
+            "Check for list public IPs response return valid data"
+        )
+
+        self.assertTrue(
+            len(public_ips) >= 1,
+            "Check public IP list has at least one IP"
+        )
+        return public_ips
+
+
+    @attr(tags=["advanced", "advancedns", "ssh"], required_hardware="true")
+    def test_router_dns_guestipquery(self):
+        """Checks that guest VM can query VR DNS"""
+
+        self.logger.debug("Starting test_router_dns_guestipquery...")
+        public_ip = self.test_router_common()[0]
+
+        self.logger.debug("Creating Firewall rule for VM ID: %s" % self.vm1.id)
+        FireWallRule.create(
+            self.apiclient,
+            ipaddressid=public_ip.id,
+            protocol=self.services["natrule1"]["protocol"],
+            cidrlist=['0.0.0.0/0'],
+            startport=self.services["natrule1"]["publicport"],
+            endport=self.services["natrule1"]["publicport"]
+        )
+
+        self.logger.debug("Creating NAT rule for VM ID: %s" % self.vm1.id)
+        nat_rule1 = NATRule.create(
+            self.apiclient,
+            self.vm1,
+            self.services["natrule1"],
+            public_ip.id
+        )
+        nat_rules = list_nat_rules(
+            self.apiclient,
+            id=nat_rule1.id
+        )
```

Comment:
This is followed by a set of assertions to ensure that valid nat rules are there.